examples/albef/data/retrieval_datamodule.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional, Tuple
+
+import torch
+from examples.albef.data.retrieval_dataset import (
+    ImageToTextRetrievalDataset,
+    RetrievalTrainingDataset,
+    TextToImageRetrievalDataset,
+)
+from examples.albef.data.transforms import (
+    ALBEFTextTransform,
+    testing_image_transform,
+    training_image_transform,
+)
+from pytorch_lightning import LightningDataModule
+from torch import Tensor
+from torch.nn.utils.rnn import pad_sequence
+from torch.utils.data import DataLoader, Dataset, DistributedSampler
+
+
+class RetrievalDataModule(LightningDataModule):
+    """
+    The Data Module for Retrieval task.
+
+    Args:
+        train_files (List[str]): The paths to training json files.
+        test_files (List[str]): The paths to testing json files.
+        image_root (str): The path to image data directory.
+        batch_size (int): The sampling batch size.
+        num_workers (int): The number of workers for the distributed mode.
+    """
+
+    def __init__(
+        self,
+        train_files: List[str],
+        test_files: List[str],
+        image_root: str,
+        batch_size: int,
+        num_workers: int,
+    ) -> None:
+        super().__init__()
+        self.train_dataset = RetrievalTrainingDataset(
+            train_files,
+            image_root,
+            training_image_transform(),
+            ALBEFTextTransform(truncate=True, max_seq_len=30, add_end_token=False),
+        )
+
+        self.image_dataset = ImageToTextRetrievalDataset(
+            test_files,
+            image_root,
+            testing_image_transform(),
+        )
+
+        self.text_dataset = TextToImageRetrievalDataset(
+            test_files,
+            ALBEFTextTransform(
+                truncate=True,
+                pad_to_max_seq_len=True,
+                max_seq_len=30,
+                add_end_token=False,
+            ),
+        )
+
+        self.batch_size = batch_size
+        self.num_workers = num_workers
+
+    def _get_sampler(
+        self,
+        dataset: Dataset,
+        shuffle: bool,
+        is_distributed: bool,
+        num_tasks: int,
+        global_rank: int,
+    ) -> Optional[DistributedSampler]:
+        if not is_distributed:
+            return None
```

Comment:
add a comment explaining this

Reply:
Thanks for adding the comment. Can you go one level deeper though? Basically as the user I want to know which sampler I will actually be using when not in distributed mode (i.e. what is the default sampler for DataLoader, in this case I think it should be `RandomSampler`)